### ISSUE TITLE
New broadcast corrupts memory?

### ISSUE THREAD

**Issue by ChrisRackauckas** [JuliaGPU/CUDA.jl#1457]
Here's an odd one.

```julia
using CUDA

const Nx = 66                                                                               # No.of steps in x-axis
const Ny = 66 
u_GPU= CUDA.zeros(Nx,Ny)
du_GPU = similar(u_GPU)     
u_GPU[25:35, 25:35] .= 50f0

#=
ERROR: CUDA error: an illegal memory access was encountered (code 700, ERROR_ILLEGAL_ADDRESS)
Stacktrace:
  [1] throw_api_error(res::CUDA.cudaError_enum)
    @ CUDA C:\Users\accou\.julia\packages\CUDA\5jdFl\lib\cudadrv\error.jl:91
  [2] macro expansion
    @ C:\Users\accou\.julia\packages\CUDA\5jdFl\lib\cudadrv\error.jl:101 [inlined]
  [3] cuOccupancyMaxPotentialBlockSize(minGridSize::Base.RefValue{Int32}, blockSize::Base.RefValue{Int32}, func::CuFunction, blockSizeToDynamicSMemSize::Ptr{Nothing}, dynamicSMemSize::Int64, blockSizeLimit::Int64)
    @ CUDA C:\Users\accou\.julia\packages\CUDA\5jdFl\lib\utils\call.jl:26
  [4] #launch_configuration#51
    @ C:\Users\accou\.julia\packages\CUDA\5jdFl\lib\cudadrv\occupancy.jl:59 [inlined]
  [5] launch_configuration
    @ C:\Users\accou\.julia\packages\CUDA\5jdFl\lib\cudadrv\occupancy.jl:56 [inlined]
  [6] #launch_heuristic#282
    @ C:\Users\accou\.julia\packages\CUDA\5jdFl\src\gpuarrays.jl:22 [inlined]
  [7] _copyto!
    @ C:\Users\accou\.julia\packages\GPUArrays\VNhDf\src\host\broadcast.jl:73 [inlined]
  [8] materialize!
    @ C:\Users\accou\.julia\packages\GPUArrays\VNhDf\src\host\broadcast.jl:51 [inlined]
  [9] materialize!(dest::SubArray{Float32, 2, CuArray{Float32, 2, CUDA.Mem.DeviceBuffer}, Tuple{UnitRange{Int64}, UnitRange{Int64}}, false}, bc::Base.Broadcast.Broadcasted{Base.Broadcast.DefaultArrayStyle{0}, Nothing, typeof(identity), Tuple{Float32}})
    @ Base.Broadcast .\broadcast.jl:868
 [10] top-level scope
    @ REPL[4]:1
=#

u_GPU= CUDA.zeros(Nx,Ny);

#=
ERROR: CUDA error: an illegal memory access was encountered (code 700, ERROR_ILLEGAL_ADDRESS)
Stacktrace:
  [1] throw_api_error(res::CUDA.cudaError_enum)
    @ CUDA C:\Users\accou\.julia\packages\CUDA\5jdFl\lib\cudadrv\error.jl:91
  [2] macro expansion
    @ C:\Users\accou\.julia\packages\CUDA\5jdFl\lib\cudadrv\error.jl:101 [inlined]
  [3] cuMemAllocAsync(dptr::Base.RefValue{CuPtr{Nothing}}, bytesize::Int64, hStream::CuStream)
    @ CUDA C:\Users\accou\.julia\packages\CUDA\5jdFl\lib\utils\call.jl:26
  [4] #alloc#1
    @ C:\Users\accou\.julia\packages\CUDA\5jdFl\lib\cudadrv\memory.jl:83 [inlined]
  [5] macro expansion
    @ C:\Users\accou\.julia\packages\CUDA\5jdFl\src\pool.jl:41 [inlined]
  [6] macro expansion
    @ .\timing.jl:359 [inlined]
  [7] actual_alloc(bytes::Int64; async::Bool, stream::CuStream)
    @ CUDA C:\Users\accou\.julia\packages\CUDA\5jdFl\src\pool.jl:39
  [8] macro expansion
    @ C:\Users\accou\.julia\packages\CUDA\5jdFl\src\pool.jl:224 [inlined]
  [9] macro expansion
    @ .\timing.jl:359 [inlined]
 [10] #_alloc#204
    @ C:\Users\accou\.julia\packages\CUDA\5jdFl\src\pool.jl:305 [inlined]
 [11] #alloc#203
    @ C:\Users\accou\.julia\packages\CUDA\5jdFl\src\pool.jl:291 [inlined]
 [12] alloc
    @ C:\Users\accou\.julia\packages\CUDA\5jdFl\src\pool.jl:287 [inlined]
 [13] CuArray{Float32, 2, CUDA.Mem.DeviceBuffer}(#unused#::UndefInitializer, dims::Tuple{Int64, Int64})
    @ CUDA C:\Users\accou\.julia\packages\CUDA\5jdFl\src\array.jl:42
 [14] CuArray
    @ C:\Users\accou\.julia\packages\CUDA\5jdFl\src\array.jl:125 [inlined]
 [15] CuArray
    @ C:\Users\accou\.julia\packages\CUDA\5jdFl\src\array.jl:136 [inlined]
 [16] CuArray
    @ C:\Users\accou\.julia\packages\CUDA\5jdFl\src\array.jl:138 [inlined]
 [17] zeros
    @ C:\Users\accou\.julia\packages\CUDA\5jdFl\src\array.jl:551 [inlined]
 [18] zeros(::Int64, ::Int64)
    @ CUDA C:\Users\accou\.julia\packages\CUDA\5jdFl\src\array.jl:553
 [19] top-level scope
    @ REPL[5]:1
=#
```

Notice that the first call to `u_GPU= CUDA.zeros(Nx,Ny)` is fine, but the second time it's called it errors with illegal memory.

**Comment by maleadt** [comment]
This is likely due to 1.8-beta1 being broken. I can't reproduce, and the illegal memory accesses are fixed on 1.8-beta2.

**Comment by ChrisRackauckas** [comment]
Indeed it is, I cannot recreate on v1.7
